### DISCUSSION TITLE
CH03_07 AWS CLI Credential 관련

### DISCUSSION THREAD

**Discussion by jeongwookie** [tedilabs/fastcampus-devops#30] [General]
안녕하세요.
강의 잘 수강하고 있습니다^^
다름 아니라 AWS CLI 설치 후 config 파일 수정하는 도중 어이없는 troubleshooting을 하게되어 디스커션에 남깁니다.
제 OS는 Mac Big Sur이며, config 파일을 제대로 작성했음에도 aws sts get-caller-identity 수행 시 아래와 같은 에러가 떴습니다.
 
An error occurred (SignatureDoesNotMatch) when calling the GetCallerIdentity operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

credential 기입 방식에 문제 있나 살펴보고, config에서 분리해서 credential 파일을 따로 만들어서 key 적어보고, aws credential 명령어 쳐서 직접 정보 기입하고.. 심지어 key를 다시 발급받아서 해봤는데도 동일한 에러가 발생했습니다.

그래서 화가나던 도중 어떤 블로거분께서 secret key에 %나 /가 포함되어 있으면 OS가 문제인건지 제대로 읽히지 않더라는 글을 발견했습니다. (https://lynlab.co.kr/blog/52)
이게 말이되나 생각도 되고 4년 전 글이라 에이 하고 제가 발급받은 2개의 key를 살펴봤는데 전부 공교롭게도 %가 포함되어 있더라구요;; 갑자기 신뢰도 상승해서 %나 / 안나올때까지 key를 발급받은 후 넣어보니 됩니다......

같은 이슈에 부딪힐 수 있는 수강생들을 위해 남겨요 ㅠㅠ

**Comment by posquit0** [comment]
안녕하세요!
좋은 정보 감사합니다.
저도 처음 듣는 이슈라 구글링을 해보니 공식 문서 상에서도 트러블슈팅 란에 해당 내용이 명시되어 있네요~ 처음부터 Secret Key의 CharSet에서 제외해도 될 것 같은데 신기하군요.

### References

https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-troubleshooting.html#tshoot-signature-does-not-match
